### PR TITLE
Render world border relative to player Y

### LINKED ISSUE #266: World Border Doesn't Display beyond vanilla height bounds
in addition, it should be make so that the worldborder forms into a box (ie, a floor of worldborder)
it could be argued that the player should simply fall into the void, but in vanilla it is straight impossible to access the void, because of bedrock, and its not unreasonable to say that the worldborder could serve the same funtion

### PR DESCRIPTION
**Pull request by DaMatrix**
This fixes #266 .

### CODE REVIEW

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal.java [hunk 1]
```diff
@@ -176,4 +181,36 @@ private RenderChunk getRenderChunkOffset(BlockPos playerPos, RenderChunk renderC
                         MathHelper.abs(playerPos.getZ() - blockpos.getZ()) > this.renderDistanceChunks * 16 ? null :
                                 this.viewFrustum.getRenderChunk(blockpos);
     }
+
+    //get render entity, can't fail (inject at HEAD
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal.java]: )

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal.java [hunk 2]
```diff
@@ -176,4 +181,36 @@ private RenderChunk getRenderChunkOffset(BlockPos playerPos, RenderChunk renderC
...
+    @Inject(method = "renderWorldBorder", at = @At(value = "HEAD"), require = 1)
+    private void getWorldFromExecute(Entity entity, float partialTicks, CallbackInfo cbi) {
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal.java]: No need for that inject, ModifyConstant [can optionally acvcept method arguments too](https://github.com/SpongePowered/Mixin/blob/d03215856d6ddf79ded288aca45fc2a0a9e2b793/src/main/java/org/spongepowered/asm/mixin/injection/ModifyConstant.java#L53-L55).

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal.java [hunk 3]
```diff
@@ -176,4 +181,36 @@ private RenderChunk getRenderChunkOffset(BlockPos playerPos, RenderChunk renderC
...
+            method = "renderWorldBorder",
+            constant = {
+                    @Constant(doubleValue = 0.0D, ordinal = 0),
+                    @Constant(doubleValue = 0.0D, ordinal = 1),
+                    @Constant(doubleValue = 0.0D, ordinal = 2),
+                    @Constant(doubleValue = 0.0D, ordinal = 3),
+                    @Constant(doubleValue = 0.0D, ordinal = 4),
+                    @Constant(doubleValue = 0.0D, ordinal = 5),
+                    @Constant(doubleValue = 0.0D, ordinal = 6),
+                    @Constant(doubleValue = 0.0D, ordinal = 7),
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal.java]: 8 ordinals!? And require=2? Why? Why not use Slice properly here?

**Review by Niko-sk2x — CHANGES_REQUESTED** [review]

**Comment by DaMatrix** [comment]
Sorry about that, never used ModifyConstant before. Should be good now

File: src/main/java/cubicchunks/asm/JvmNames.java [hunk 4]
```diff
@@ -82,6 +83,7 @@ private JvmNames() {
         MATERIAL_WATER = MATERIAL + "WATER:" + MATERIAL,
         MUTABLE_BLOCK_POS_CONSTRUCT = MUTABLE_BLOCK_POS + "<init>(III)V",
         POOLED_MUTABLE_BLOCK_POS_SET_POS_D = POOLED_MUTABLE_BLOCK_POS + "setPos(DDD)" + POOLED_MUTABLE_BLOCK_POS,
+        TESSELLATOR_DRAW = TESSELLATOR + "draw()V",
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/JvmNames.java]: I didn't notice it before, but don't use JvmNames anymore, it's deprecated for a reason. It's from before Intellij + MinecraftDev plugin had autocomplete for this stuff.

Comment (DaMatrix) [reply src/main/java/cubicchunks/asm/JvmNames.java]: Alright, so should I just put the method names directly into the mixin annonations?

Comment (Niko-sk2x) [reply src/main/java/cubicchunks/asm/JvmNames.java]: yes. As MinecraftDev plugin for intellij has good autocomplete for it, there is no reason to use such constants for it. It just makes things harder to change, and harder to get help when you get something wrong

**Review by Niko-sk2x — APPROVED** [review]

**Comment by Niko-sk2x** [comment]
Just waiting for travis now...

**Comment by Niko-sk2x** [comment]
Fix imports

**Pull request merged**